components/ReleaseStats/ReleaseStats.jsx

```diff
@@ -0,0 +1,163 @@
+        .from("codes")
+        .select("redeemed_at", { count: "exact", head: true })
+        .eq("release_id", release.id)
+        .eq("redeemed", true)
+        .gte("redeemed_at", startDate)
+        .lte("redeemed_at", endDate)
+
+      if (count) {
+        setRedeemedCodes(count)
+      } else if (count === 0) {
+        setRedeemedCodes(0)
+      }
+
+      if (error) throw error
+    } catch (error) {
+      throw error
+    }
+  }
+
+  function closeModal() {
+    setOpen(false)
+  }
+  return (
+    <Dialog open={open} onOpenChange={setOpen}>
+      <DialogTrigger
+        className="button"
+        data-variant="secondary"
+        data-size="small"
+      >
+        Stats
+      </DialogTrigger>
+
+      <DialogContent>
+        <header>
+          <h2 className="text-3">Release Statistics</h2>
+        </header>
+
+        <div className="stack">
+          <h3 className="text-3">{release.title} Codes</h3>
+          <p>Total codes redeemed: {redeemedCodes}</p>
+          <div className="flex-grid">
+            <button
+              className="button"
+              data-variant="primary"
+              onClick={() =>
+                getCodes(
+                  today.toLocaleDateString(),
+                  tomorrow.toLocaleDateString()
+                )
+              }
+            >
+              Today
+            </button>
+
+            <button
+              className="button"
+              data-variant="primary"
+              onClick={() =>
+                getCodes(
+                  yesterday.toLocaleDateString(),
+                  today.toLocaleDateString()
+                )
+              }
+            >
+              Yesterday
+            </button>
+
+            <button
+              className="button"
+              data-variant="primary"
+              onClick={() =>
+                getCodes(
+                  pastWeek.toLocaleDateString(),
+                  tomorrow.toLocaleDateString()
+                )
+              }
+            >
+              7 Days
+            </button>
+            <button
+              className="button"
+              data-variant="primary"
+              onClick={() =>
+                getCodes(
+                  pastTwoWeeks.toLocaleDateString(),
+                  tomorrow.toLocaleDateString()
+                )
+              }
+            >
+              14 Days
+            </button>
+            <button
+              className="button"
+              data-variant="primary"
+              onClick={() =>
+                getCodes(
+                  pastMonth.toLocaleDateString(),
+                  tomorrow.toLocaleDateString()
+                )
+              }
+            >
+              30 Days
+            </button>
+            <button
+              className="button"
+              data-variant="primary"
+              onClick={() =>
+                getCodes("2023-01-22", tomorrow.toLocaleDateString())
```

Comment:
Why is this a static date?

Reply:
Its just a date from before the database existed to make sure that it gets all of the dates.